rust/shalom_core/src/operation/types.rs

```diff
@@ -100,3 +108,11 @@ impl EnumSelection {
         })
     }
 }
+
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct VariableDefinition {
+    pub name: String,
+    pub ty_name: String,
+    pub is_optional: bool,
+    pub default_value: Option<String>,
+}
```

Comment:
ty_name is ok though it should be possible to know what is the type of this variable.
so 
```suggestion
#[derive(Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
pub struct VariableDefinition {
pub name: String,
pub(crate) ty: schema::types::GraphQLAny
    pub is_optional: bool,
    pub default_value: Option<String>,
}
```